src/libaktualizr/uptane/directorrepository.cc

```diff
@@ -198,4 +198,24 @@ void DirectorRepository::dropTargets(INvStorage& storage) {
   resetMeta();
 }
 
+bool DirectorRepository::matchTargetsWithImageTargets(const Uptane::Targets& image_targets) const {
+  // step 10 of https://uptane.github.io/papers/ieee-isto-6100.1.0.0.uptane-standard.html#rfc.section.5.4.4.2
+  // TBD: no delegation support, consider reusing of findTargetInDelegationTree()
+  // that needs to be moved into a common place to be resued by Primary and Secondary
```

Comment:
I was just this week looking into how hard that would be. Unfortunately, it isn't simple. But I still think we need to do it sooner or later.

Also, currently, this function is only used by `AktualizrSecondary`. Why does this need to live here? Or can it be reused by the Primary?

Reply:
> Or can it be reused by the Primary?
Yes.

Reply:
>> Or can it be reused by the Primary?
>
> Yes.

Great. If that's easy, can you do that now? Otherwise, leave a TODO specifically for that we can come back for it later.

Reply:
not trivial as it might look at first glance, adding TODO.